packages/e2e-tests/specs/experiments/blocks/comments-query.test.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	activateTheme,
+	clickBlockAppender,
+	createNewPost,
+	getEditedPostContent,
+	insertBlock,
+	setOption,
+} from '@wordpress/e2e-test-utils';
+
+describe( 'Comment Query Loop', () => {
+	beforeAll( async () => {
+		await activateTheme( 'emptytheme' );
+		await setOption( 'page_comments', true, 'options-discussion.php' );
+		await setOption( 'comments_per_page', '2', 'options-discussion.php' );
+		await createNewPost();
+	} );
+	it( 'Comment Query block insertion', async () => {
+		// We won't select the option that we updated and will also remove some
+		// _transient options that seem to change at every update.
+		await insertBlock( 'Comments Query Loop' );
+		expect( await getEditedPostContent() ).toMatchSnapshot();
+	} );
```

Comment:
While it gives some confidence that the block can be inserted, I'm a bit skeptical if we should have a test that really only checks for that. Each test means additional maintenance (when the implementation changes, and the test has to be updated accordingly).

This is especially true with snapshot tests; while convenient, snapshot tests aren't very "semantic" in expressing our expectations. And it's quite possible that the exact block markup changes frequently (e.g. a newline is removed or introduced -- changes that wouldn't even have any impact on the block behavior), and there's a risk that we get used to updating them blindly and start ignoring the actual snapshot contents.

Another question would be why the Comments Query block needs a test to cover its insertion more than other blocks (since not all of them test for that). We've seen one case when the block was gone from the inserter (#39415), so that's a fair argument.

Still, TBH, I'd remove the snapshot test. IMO, it'd be best to have an e2e test that covers some behavior that's fairly specific to the block, and ideally has already seen a regression; a test like that would require block insertion anyway, so we'd get test coverage for that "for free". (I thought that we needed a test for the [scenario described here](https://github.com/WordPress/gutenberg/pull/39227#issuecomment-1067922582), but it seems like you've covered that now with a unit test?)

Reply:
It is covered by a unit test, but, as you mention, is not so clear what is that test solving.
For that reason, maybe creating an e2e of this case scenario would be a better idea, the only cons is that we have to create a lot of new funcionalities:

 - Setting and getting discussion options (almost done)
 - Creating at least 3 comments on the post/page.
 - Checking different elements on the frontend.

Reply:
I think that might be okay 🤔 Since this is covering a whole new area of Gutenberg (the Comments blocks), it makes sense that it requires us to introduce corresponding e2e helpers.

> Setting and getting discussion options (almost done)

👍 🎉 

> Creating at least 3 comments on the post/page.

I guess we can start with a helper that creates one comment and allows to specify whether it's a reply to the post, or to a different comment.

We should probably ask ourselves whether we want to:
1. Create those comments via the frontend, which will implicitly also test the frontend functionality of those blocks. It means that any e2e test relying on the util will break if any of those blocks break (probably a good thing), or change their behavior, in which case the util will need updating (potentially less good, since it means more maintenance); or
2. Use the `wp-admin` Comments panel (i.e. `wp-admin/edit-comments.php`). This is going to be stabler as we don't expect it to change as much as Gutenberg and the new Comments blocks. But maybe we really want the "Create Comment" helper to work via the frontend, especially if we're adding more features that we want to cover? 🤔 

Curious to hear your thoughts!

> Checking different elements on the frontend.

There _might_ be some precedent for this (but certainly not a lot), since it's a problem that doesn't just apply to the Comments blocks, but to FSE more generally. Maybe try looking in the Site Editor e2e tests directory.

Reply:
> Use the wp-admin Comments panel (i.e. wp-admin/edit-comments.php). This is going to be stabler as we don't expect it to change as much as Gutenberg and the new Comments blocks. But maybe we really want the "Create Comment" helper to work via the frontend, especially if we're adding more features that we want to cover? 

I'm afraid there is no option to create comments from there 😢 . So I guess we have to go with option 1 (creating on the front).

Reply:
> I'm afraid there is no option to create comments from there 😢 . So I guess we have to go with option 1 (creating on the front).

Ah, oops 😅  Looks like I didn't pay close enough attention! Well, fewer decisions for us to make 😅